sklearn/tests/test_common.py

```diff
@@ -297,3 +304,186 @@ def test_search_cv(estimator, check, request):
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError
+        proba_shape = (X.shape[0], self.classes_.size)
+        y_proba = np.zeros(shape=proba_shape, dtype=np.float64)
+        y_proba[:, self._most_frequent_class_idx] = 1.0
+        return y_proba
+
+    def predict(self, X):
+        y_proba = self.predict_proba(X)
+        y_pred = y_proba.argmax(axis=1)
+        return self.classes_[y_pred]
+
+    def score(self, X, y):
+        from sklearn.metrics import accuracy_score
+        return accuracy_score(y, self.predict(X))
+
+
+class MinimalRegressor:
+    """Minimal regressor implementation with inheriting from BaseEstimator."""
+    _estimator_type = "regressor"
+
+    def __init__(self, param=None):
+        self.param = param
+
+    def __repr__(self):
+        # Only required when using pytest-xdist to get an id not associated
+        # with the memory location
+        return self.__class__.__name__
+
+    def get_params(self, deep=True):
+        return {"param": self.param}
+
+    def set_params(self, **params):
+        for key, value in params.items():
+            setattr(self, key, value)
+        return self
+
+    def __getstate__(self):
+        return self.__dict__.copy()
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def fit(self, X, y):
+        X, y = check_X_y(X, y)
+        self.n_features_in_ = X.shape[1]
+        self._mean = np.mean(y)
+        return self
+
+    def predict(self, X):
+        check_is_fitted(self)
+        X = check_array(X)
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError
+        return np.ones(shape=(X.shape[0],)) * self._mean
+
+    def score(self, X, y):
+        from sklearn.metrics import r2_score
+        return r2_score(y, self.predict(X))
+
+
+class MinimalTransformer:
+    """Minimal transformer implementation with inheriting from
+    BaseEstimator."""
+
+    def __init__(self, param=None):
+        self.param = param
+
+    def __repr__(self):
+        # Only required when using pytest-xdist to get an id not associated
+        # with the memory location
+        return self.__class__.__name__
+
+    def get_params(self, deep=True):
+        return {"param": self.param}
+
+    def set_params(self, **params):
+        for key, value in params.items():
+            setattr(self, key, value)
+        return self
+
+    def __getstate__(self):
+        return self.__dict__.copy()
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def fit(self, X, y=None):
+        X = check_array(X)
+        self.n_features_in_ = X.shape[1]
+        return self
+
+    def transform(self, X, y=None):
+        check_is_fitted(self)
+        X = check_array(X)
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError
+        return X
+
+    def fit_transform(self, X, y=None):
+        return self.fit(X, y).transform(X, y)
+
+
+def _generate_minimal_compatible_instances():
```

Comment:
Why do we wrap them in a SearchCV? should this be expalined in the docstring?

Reply:
Yes I don't think this method should wrap them by default.

Instead I think that we remove should remove `test_minimal_class_implementation_checks` from `test_common.py` add a new test in `test_estimator_checks.py` that checks that `check_estimator` work on those 3 classes (without using `parametrize_with_checks` and without using a generator for yielding those minimal classes, just a Python list is enough).

Furthermore, we should have some new test in:

- test_pipeline.py to check that they can be in pipelines;
- test_search_cv.py to check that they can be used in a grid search.

Reply:
Having the test in `test_estimator_checks.py` will fail with minimum dependence builds (we don't ask for pytest)

Reply:
Sorry I did not see that you mentioned not to use `parametrize_with_checks`.